internal/namespace/caveats.go

```diff
@@ -0,0 +1,57 @@
+package namespace
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/authzed/spicedb/pkg/caveats"
+	caveattypes "github.com/authzed/spicedb/pkg/caveats/types"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+/**
+ * ValidateCaveatDefinition validates the parameters and types within the given caveat
+ * definition, including usage of the parameters.
+ */
+func ValidateCaveatDefinition(caveat *core.CaveatDefinition) error {
+	if len(caveat.ParameterTypes) == 0 {
+		return newTypeErrorWithSource(
```

Comment:
Technically, we can't produce caveats that are broken in these ways. These are simply sanity checks in case we somehow do. Do you want me to write those tests anyway?

Reply:
Testing these bits would increase confidence we don't end up with weird cascading failures that are difficult to understand, even if things are supposed to work in a way _technically_. It didn't look like it was difficult to add these tests, but I'll leave it up to you.